app/packs/src/apps/moleculeModerator/MoleculeModerator.js

```diff
@@ -95,48 +69,90 @@ class MoleculeModerator extends Component {
       });
   }
 
-  handleEditorSave(molfile, svg_file = null, config = null) {
+  handleEditorSave(molfile, svgFile = null, config = null) {
     const { molecule } = this.state;
     molecule.molfile = molfile;
     const smiles = config ? config.smiles : null;
     const isChemdraw = !!smiles;
-    MoleculesFetcher.renewSVGFile(molecule.id, svg_file, isChemdraw).then((json) => {
+    MoleculesFetcher.renewSVGFile(molecule.id, svgFile, isChemdraw).then((json) => {
       molecule.molecule_svg_file = json.svg_path;
       this.setState({ molecule, showStructureEditor: false });
     });
   }
 
+  handleSearchTermChange(e) {
+    this.setState((state) => ({
+      ...state,
+      searchTerm: e.target.value.trim(),
+    }));
+  }
+
+  onSave() {
+    LoadingActions.start();
+    let { molecule } = this.state;
+    const { msg } = this.state;
+    MoleculesFetcher.updateMolfileSVG(molecule).then((json) => {
+      if (json.msg.level !== 'error') {
+        ({ molecule } = json);
+      }
+      msg.show = true;
+      msg.level = json.msg.level;
+      msg.message = json.msg.message;
+      this.setState({ molecule, msg, showStructureEditor: false });
+      LoadingActions.stop();
+      pageNotify('Save', msg.level, msg.message);
+    });
+  }
+
   render() {
+    const { molecule, showStructureEditor, searchTerm } = this.state;
+
     const formSearch = (
-      <form>
-        <FormGroup controlId="frmCtrlInChiKey">
-          <ControlLabel>InChiKey</ControlLabel>
-          <FormControl type="text" placeholder="Enter text" inputRef={(ref) => { this.refInChiKey = ref; }} />
-        </FormGroup>
-        <Button onClick={this.handleSearch}>Search&nbsp;<i className="fa fa-search" aria-hidden="true" /></Button>
-      </form>
+      <Form>
+        <Form.Group controlId="frmCtrlInChiKey">
+          <Form.Label>InChiKey</Form.Label>
+          <Form.Control
+            className="mb-2"
+            type="text"
+            placeholder="Enter text"
+            value={searchTerm}
+            onChange={this.handleSearchTermChange}
+          />
+        </Form.Group>
+        <Button onClick={this.handleSearch}>
+          Search
+          <i className="fa fa-search ms-1" aria-hidden="true" />
+        </Button>
+      </Form>
     );
 
-    const pageComponent = this.state.molecule ?
-      (<MoleculeModeratorComponent
-        molecule={this.state.molecule}
-        showStructureEditor={this.state.showStructureEditor}
-        handleEditorSave={this.handleEditorSave}
-        handleEditor={this.handleEditor}
-        onSave={this.onSave}
-      />) : <div />;
-
     return (
-      <div>
-        {pageNav}
-        <div className="container" style={{ marginTop: '60px' }}>
-          {formSearch}
-          <hr />
-        </div>
-        {pageComponent}
+      <>
+        <Navbar className="navbar-custom px-4">
+          <Nav className="container-md justify-content-start">
```

Comment:
justify-content-start is default behavior and does not need to be applied explicitly.